code/API_definitions/sim-swap-notification-subscription.yaml

```
@@ -0,0 +1,615 @@
+
+paths:
+  /subscriptions:
+    post:
+      tags:
+        - Sim swap notification subscription
+      summary: 'Create a sim swap event subscription for a phone number'
+      description: Create a sim swap event subscription for a phone number
+      operationId: createSimSwapSubscription
+      security:
+        - openId:
+          - sim-swap:subscriptions:create
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateSubscription'
+        required: true
+      callbacks:
+        notifications:
+          "{$request.body#/webhook/notificationUrl}":
+            post:
+              tags:
+                - Session notifications callback
+              summary: "Session notifications callback"
+              description: |
+                Important: this endpoint is to be implemented by the API consumer.
+                The sim swap server will call this endpoint whenever a swapped event occurs.
+              operationId: postNotification
+              requestBody:
+                required: true
+                content:
+                  application/json:
+                    schema:
+                      $ref: "#/components/schemas/CloudEvent"
+                    examples:
+                      SWAPPED:
+                        $ref: '#/components/examples/SWAPPED'
+                      SUBSCRIPTION_ENDS:
+                        $ref: '#/components/examples/SUBSCRIPTION_ENDS'  
+              responses:
+                "204":
+                  description: Successful notification
+                "400":
+                  $ref: "#/components/responses/Generic400"
+                "401":
+                  $ref: "#/components/responses/Generic401"
+                "403":
+                  $ref: "#/components/responses/Generic403"
+                "500":
+                  $ref: "#/components/responses/Generic500"
+                "503":
+                  $ref: "#/components/responses/Generic503"
+              security:
+                - { }
+                - notificationsBearerAuth: [ ]
+
+      responses:
+        201:
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SubscriptionInfo'
+        202 :
+          description: Request accepted to be processed. It applies for async creation process.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SubscriptionAsync'
+        400:
+          $ref: '#/components/responses/Generic400'
+        401:
+          $ref: '#/components/responses/Generic401'
+        403:
+          $ref: '#/components/responses/Generic403'
+        409:
+          $ref: '#/components/responses/Generic409'
+        500:
+          $ref: '#/components/responses/Generic500'
+        503:
+          $ref: '#/components/responses/Generic503'

```

The HTTP error codes must be in quotes. [Source](https://swagger.io/specification/v3/#responses-object)

```suggestion
        "201":
          description: Created
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SubscriptionInfo'
        "202":
          description: Request accepted to be processed. It applies for async creation process.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/SubscriptionAsync'
        "400":
          $ref: '#/components/responses/Generic400'
        "401":
          $ref: '#/components/responses/Generic401'
        "403":
          $ref: '#/components/responses/Generic403'
        "409":
          $ref: '#/components/responses/Generic409'
        "500":
          $ref: '#/components/responses/Generic500'
        "503":
          $ref: '#/components/responses/Generic503'

```